app/domain/authentication/constraints/at_least_one_constraint.rb

```diff
@@ -0,0 +1,12 @@
+module Authentication
+  module Constraints
+
+    # This constraint returns true if the resource restriction contains at least one restriction. Otherwise it raises
+    # EmptyAnnotationsListConfigured exception
+    class AtLeastOneConstraint
+      def validate(resource_restrictions:)
+        raise Errors::Authentication::Constraints::EmptyAnnotationsListConfigured unless resource_restrictions.any?
```

Comment:
negation better not be used if not necessary.
`ruby` contains negation for many checkings:
```suggestion
        raise Errors::Authentication::Constraints::EmptyAnnotationsListConfigured if resource_restrictions.empty?
```